frontend/packages/pipelines-plugin/src/components/shared/common/auto-complete/AutoCompletePopover.tsx

```diff
@@ -14,25 +16,74 @@ const AutoCompletePopover: React.FC<AutoCompletePopoverProps> = ({
   children,
   onAutoComplete,
 }) => {
-  const { contentRef, popperProps, options, insertAutoComplete } = useAutoComplete(
-    autoCompleteValues,
-    onAutoComplete,
-  );
+  const { t } = useTranslation();
+
+  const {
+    contentRefCallback,
+    focusTrapProps,
+    insertAutoComplete,
+    popperProps,
+    menuWidth,
+    options,
+  } = useAutoComplete(autoCompleteValues, onAutoComplete);
 
   return (
     <>
-      {children(contentRef)}
-      <Popper placement="bottom-start" closeOnEsc closeOnOutsideClick {...popperProps}>
-        <Menu onSelect={(event, itemId: number) => insertAutoComplete(options[itemId])}>
-          <MenuList translate="no">
-            {options.map((value, idx) => (
-              <MenuItem key={value} itemId={idx} translate="no">
-                {value}
-              </MenuItem>
-            ))}
-          </MenuList>
-        </Menu>
-      </Popper>
+      {children(contentRefCallback)}
+      <WithScrollContainer>
+        {(scrollContainer) => (
+          <Popper
+            {...popperProps}
+            placement="bottom-start"
+            closeOnEsc
+            closeOnOutsideClick
+            container={scrollContainer}
+            popperOptions={{
+              modifiers: {
+                hide: {
+                  enabled: false,
+                },
+                preventOverflow: {
+                  enabled: false,
+                },
+                flip: {
+                  enabled: false,
+                },
+              },
+            }}
+          >
+            <FocusTrap {...focusTrapProps}>
+              <Menu
+                style={{ maxHeight: 200, overflow: 'auto', width: menuWidth }}
+                onSelect={(event, itemId: number) => {
+                  const text = options[itemId];
+                  if (text) {
+                    insertAutoComplete(text);
+                  }
+                }}
+              >
+                <MenuList translate="no">
+                  {options.length === 0 ? (
+                    // There is a tab-index problem with the Menus from PF. If the first option is removed, it breaks tab indexes
```

Comment:
@karthikjeeyar This issue here is why we can't tab. The tab index for all but the first item is `-1`... why? 🤷